### PR TITLE
bugfix: `ValueObject` and `URL`

### LINKED ISSUE #94: If nesting struct 2 times or more, the parser throws error
The following will be error.

```swift
struct A {
  let string: String
}

struct B {
  let a: A
  let string: String
}

struct C {
  let b: B
}

let target = C(b: B(a: A(string: "hoge"), string: "a"))
Debug.pp >>> target
```

the following is error message.

```swift
Unknown error:
▿ SwiftPrettyPrintTests.MultilineFormatterTests.B
  ▿ a: SwiftPrettyPrintTests.MultilineFormatterTests.A
    - string: "hoge"
  - string: "a"
```

### FYI
The following will **not** be error.

```swift
struct A {
  let string: String
}

struct B {
  let a: A
  let string: String
}

let target = B(a: A(string: "hoge"), string: "a")
Debug.pp >>> target
```

### PR DESCRIPTION
**Pull request by YusukeHosonuma**
and many refactors :)

resolved #94 

## TODO
- [x] change destination to `master`
- [x] bugfix
  - [x] #94 
  - [x] value of type the `URL` is not print type information when `debug` is `true`
- [x] refactors

### CODE REVIEW

File: Sources/Core/Pretty.swift [hunk 1]
```diff
@@ -16,17 +16,21 @@ struct Pretty {
...
-                return _value(target)
+                if let value = mirror.children.first?.value {
+                    if debug {
+                        return "Optional(" + _string(value) + ")"
+                    } else {
+                        return _string(value)
+                    }
+                } else {
+                    return "nil"
+                }
```
Comment (YusukeHosonuma) [inline Sources/Core/Pretty.swift:33]: Explicitly handle `Optional` type.

File: Sources/Core/Pretty.swift [hunk 2]
```diff
@@ -74,23 +78,14 @@ struct Pretty {
             }
         }
 
-        // Swift.URL
```
Comment (YusukeHosonuma) [inline Sources/Core/Pretty.swift:77]: `URL` and `Empty` are handle in `asPremitiveString()`.

File: Sources/Core/Pretty.swift [hunk 3]
```diff
@@ -100,41 +95,6 @@ struct Pretty {
...
-                }
-            } else {
-                return "nil"
-            }
-
-        default:
-            throw PrettyError.notSupported(target: target)
-        }
-    }
-
```
Comment (YusukeHosonuma) [inline Sources/Core/Pretty.swift:137]: The function `valueString()` was removed that ambiguous process in it :)

File: Sources/Core/Pretty.swift [hunk 4]
```diff
@@ -159,6 +119,43 @@ struct Pretty {
...
+
+        case let url as URL:
+            if debug {
+                return #"URL("\#(url.absoluteString)")"#
+            } else {
+                return url.absoluteString
+            }
+
+        default:
+            // TODO: support other premitive type that not has child
```
Comment (YusukeHosonuma) [inline Sources/Core/Pretty.swift:160]: We should support other type such as `Long`, `Range`, `Date` and others maybe.

File: Tests/SwiftPrettyPrintTests/Public/DebugTests.swift [hunk 5]
```diff
@@ -114,7 +114,7 @@ class DebugTests: XCTestCase {
                              name: Optional("ポチ"),
                              nickname: nil,
                              age: 3,
-                             homepage: Optional(https://www.google.com/)
+                             homepage: Optional(URL("https://www.google.com/"))
```
Comment (YusukeHosonuma) [inline Tests/SwiftPrettyPrintTests/Public/DebugTests.swift:117]: bugfix: `Optional<URL>` should print with type information when `debug` is true.

It's an old bug 😅

**Review by po-miyasaka — COMMENTED** [review]
lgtm!!!

**Review by po-miyasaka — APPROVED** [review]

**Comment by sahara-ooga** [comment]
@YusukeHosonuma 
is `asValueString` method for "ValueObject"?

**Review by sahara-ooga — COMMENTED** [review]
LGTM

**Pull request merged**